### PR TITLE
[LCGS] Make IntermediateLCGS impl GameStructure

### LINKED ISSUE #28: [LCGS] Symbol table and symbol checking phase

### PR DESCRIPTION
**Pull request by NicEastvillage**
Make `IntermediateLCGS` impl `GameStructure` and other improvements to run.

Depends on #44

Resolves #28

### CODE REVIEW

**Comment by NicEastvillage** [comment]
Asger's test helped me find an error. Expect one more commit to fix `state_from_index()`.

**Review by cogitantium — APPROVED** [review]
LGTN

**Review by jenrik — CHANGES_REQUESTED** [review]
I haven't fully understond `intermediate.rs` but it seem right.

Also there is `clippy` warnings that needs to be fixed.

File: src/lcgs/ir/intermediate.rs [hunk 1]
```diff
@@ -62,32 +56,113 @@ impl IntermediateLCGS {
...
+            if let DeclKind::StateVar(var) = &self
+                .symbols
+                .get(owner, name)
+                .unwrap()
+                .declaration
+                .borrow()
+                .kind
+            {
+                let value = {
+                    let size = var.ir_range.len() as i32;
```
[the remark this answers is not in the document]
Comment (NicEastvillage) [reply src/lcgs/ir/intermediate.rs:79]: I use i32 for runtime values. The u32 and usize comes from your representation of states, players, and labels in the algorithm.

**Review by jenrik — APPROVED** [review]

**Pull request merged**